### PR TITLE
Add TaskConfig to simplify createSwiftTask calls

### PR DESCRIPTION
**Pull request by adam-fowler**
I keep seeing reasons to add more configuration options to `createSwiftTask`. Instead of adding a new function parameter every time I need a new config option. I've create a TaskConfig interface class

I'm not actually using this at the moment but will be in the future

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]
LGTM

File: src/SwiftTaskProvider.ts [hunk 1]
```diff
@@ -66,33 +89,41 @@ function createUpdateTask(): vscode.Task {
...
-        new vscode.ShellExecution(command, args)
+        new vscode.ShellExecution(command, args),
+        config?.problemMatcher
     );
     // This doesn't include any quotes added by VS Code.
     // See also: https://github.com/microsoft/vscode/issues/137895
     task.detail = `${command} ${args.join(' ')}`;
-    task.group = group;
+    task.group = config?.group;
+    if (config?.presentationOptions !== undefined) {
```
Comment (svanimpe) [inline src/SwiftTaskProvider.ts:105]:
```suggestion
    task.presentationOptions = config?.presentationOptions ?? { };
```

File: src/SwiftTaskProvider.ts [hunk 2]
```diff
@@ -66,33 +89,41 @@ function createUpdateTask(): vscode.Task {
...
-        vscode.TaskScope.Workspace,
+        config?.scope ?? vscode.TaskScope.Workspace,
         name,
         'swift',
         new vscode.ShellExecution(command, args),
-        problemMatchers
+        config?.problemMatcher
     );
+    task.group = config?.group;
+    if (config?.presentationOptions !== undefined) {
```
Comment (svanimpe) [inline src/SwiftTaskProvider.ts:124]:
```suggestion
    task.presentationOptions = config?.presentationOptions ?? { };
```

**Comment by svanimpe** [comment]
We should also update `resolveTask` to copy over any `scope`, `problemMatchers`, or `presentationOptions` from `task` to `newTask`.

**Review by svanimpe — APPROVED** [review]

**Pull request merged**